src/main/java/org/opensearch/knn/index/quantizationService/KNNVectorQuantizationTrainingRequest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.quantizationService;
+
+import org.opensearch.knn.index.vectorvalues.KNNVectorValues;
+import org.opensearch.knn.quantization.models.requests.TrainingRequest;
+
+import static org.apache.lucene.search.DocIdSetIterator.NO_MORE_DOCS;
+
+/**
+ * KNNVectorQuantizationTrainingRequest is a concrete implementation of the abstract TrainingRequest class.
+ * It provides a mechanism to retrieve float vectors from the KNNVectorValues by document ID.
+ */
+final class KNNVectorQuantizationTrainingRequest<T> extends TrainingRequest<T> {
+
+    private final KNNVectorValues<T> knnVectorValues;
+    private int lastIndex;
+
+    /**
+     * Constructs a new QuantizationFloatVectorTrainingRequest.
+     *
+     * @param knnVectorValues the KNNVectorValues instance containing the vectors.
+     */
+    KNNVectorQuantizationTrainingRequest(KNNVectorValues<T> knnVectorValues) {
+        super((int) knnVectorValues.totalLiveDocs());
```

Comment:
why you are casting this to int?

Reply:
Please don't resolve the comment without adding any response whether you are going to make the change or not.

Reply:
Sampler needs total number of Numbers which is int .. @

Reply:
All the iterators(BDV, FVV, BVV) gives long as the totalLiveDocs. Don't you think we should use long here?

Reply:
This casting has to be done somewhere , for Sampler it's just an total number of items which needs to be sampled , I think int should be fine.

Reply:
I think we do need to change this - if num vecs > 2B, I think we can run into issues. Im fine adding as a follow up in another review, but we should represent total number of vectors in an index as long because 2B is somewhat reasonable.

Reply:
This is from Lucene 

[https://github.com/apache/lucene/blob/91f45b53e6b4a9b68d110ee128797e74a6578379/luc[…]c/java/org/apache/lucene/util/quantization/ScalarQuantizer.java](https://github.com/apache/lucene/blob/91f45b53e6b4a9b68d110ee128797e74a6578379/lucene/core/src/java/org/apache/lucene/util/quantization/ScalarQuantizer.java#L306-L325)

and this is https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/DocIdSetIterator.java#L43
[DocIdSetIterator.java](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/DocIdSetIterator.java)



if you see this link https://issues.apache.org/jira/browse/LUCENE-4607 
because the current scorer/disi interface can never return anything > Integer.MAX_VALUE, so the estimatedDocCount can never be 64 bits.

I think we should change estimation to int rather than long.